05/implement-network-delay.instructor.md

```diff
@@ -1,5 +1,49 @@
 
+Sample clarifying questions:
+
+Note that for strictly defined challenges, there may not be many points that feel unclear from a specification perspective. However, we can still take the opportunity for clarification to consider what the impact of certain requirements we have been given will have on our implementation. Don't hesitate to observe something that has been specified, restate it in your own words to show your understanding, then go on to postulate why that information was given and how it might impact your approach.
+
+1. In which way is the graph being represented?
+
+   From the parameter descriptions, `times` is described as a list whose elements represented directed edges of the format `(from, to, cost)`. This means the supplied graph representation is expected to take the form of a list of edges, which we are also able to verify from the examples.
+
+   Observing this is useful, as it lets us consider whether we want to convert the list of edges into some other format that will allow us to approach the challenge more efficiently overall. In this case, we know that we'll need to efficiently iterate over the neighbors of a given node, so we'll want to convert the list of edges into an adjacency dict, allowing us to look up the neighbors using the node value as the key.
+
+2. Do the nodes start at 0 or 1?
+
+   The problem defines the nodes as being labeled from 1 to n, and the examples support this, showing nodes starting at 1. So we will work with this as a requirement.
+
+   This is useful to consider as it impacts how we relate our node data (1-based) to supplemental data (such as the total cost list, or potentially a previous list if we were asked to return path data). If we use a list to track the total cost to reach each node, we'll need to be careful about how we index into the list, either adjusting the node value to be 0-based or inserting a dummy value to effectively make the list 1-based. In the latter case, we would also need to ensure that the dummy value didn't interfere with our logic for determining the maximum cost.
+
+3. Will the graph ever be empty? What should the result be?
+
+   Since the data is being presented as a list of edges, there is a dta configuration that might feel like a natural representation of an empty graph (a graph having neither nodes nor edges): an empty list. However, technically, we can really only tell that there are no edges in such a graph. There could be any number of nodes, simple that share no edges. So we can't really tell whether the graph is empty or not. That is, we cannot tell the difference between a graph with no nodes and a graph with potentially many unconnected nodes.
+   
+   However, we are also given the count of nodes int he graph as a separate input, which does allow us to differentiate. A truly empty graph will have an empty list of edges and a count of zero nodes. A graph with no edges will have an empty list of edges and a count of some non-zero number of nodes.
```

Comment:
```suggestion
   However, we are also given the count of nodes in the graph as a separate input, which does allow us to differentiate. A truly empty graph will have an empty list of edges and a count of zero nodes. A graph with no edges will have an empty list of edges and a count of some non-zero number of nodes.
```

Reply:
What can I say? My fingers are trained to declare integer variables! I make this typo ALL THE TIME. 😆